app/src/lib/stats/stats-store.ts

```diff
@@ -28,6 +28,18 @@ import { MultiCommitOperationKind } from '../../models/multi-commit-operation'
 import { getNotificationsEnabled } from '../stores/notifications-store'
 import { isInApplicationFolder } from '../../ui/main-process-proxy'
 
+type PullRequestReviewStatFieldInfix =
+  | 'Approved'
+  | 'ChangesRequested'
+  | 'Commented'
+
+type PullRequestReviewStatFieldSuffix =
+  | 'NotificationCount'
+  | 'NotificationClicked'
+  | 'DialogSwitchToPullRequestCount'
+
+type PullRequestReviewStatField = `pullRequestReview${PullRequestReviewStatFieldInfix}${PullRequestReviewStatFieldSuffix}`
```

Comment:
This is basically the "magic" I mentioned. Basically, using the [template literal types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) to build all the relevant stat field names into a type, the rest is in the `recordBlahBlah` methods that use this type to make sure we're building the right metric key.

Reply:
It seems prettier doesn't like template literal types right now 😅  https://github.com/desktop/desktop/runs/5626068009?check_suite_focus=true/#step:8:1